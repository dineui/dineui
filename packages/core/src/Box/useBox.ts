export interface UseBoxParams {
  xs?: {}
  s?: {}
  m?: {}
  l?: {}
  xl?: {}
  xxl?: {}
  xxxl?: {}
}

export function useBox(params: UseBoxParams) {
  return {
    css: {},
  }
}
