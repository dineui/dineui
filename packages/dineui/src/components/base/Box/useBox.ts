import { CSSObject } from '@emotion/react'

export interface UseBoxParams {
  xs?: CSSObject
  s?: CSSObject
  m?: CSSObject
  l?: CSSObject
  xl?: CSSObject
  xxl?: CSSObject
  xxxl?: CSSObject
}

export const mediaBreakpoints = {
  xsmall: 0,
  small: 414,
  medium: 960,
  large: 1200,
  xlarge: 1440,
  xxlarge: 1920,
  xxxlarge: 2560,
}

export const mediaQuery = (breakpoint: number) => {
  return `@media (min-width: ${breakpoint}px)`
}

export function useBox(params: UseBoxParams = {}): any {
  const { xs, s, m, l, xl, xxl, xxxl, ...otherProps } = params

  const interStyle = {
    ...xs,
    [mediaQuery(mediaBreakpoints.small)]: s,
    [mediaQuery(mediaBreakpoints.medium)]: m,
    [mediaQuery(mediaBreakpoints.large)]: l,
    [mediaQuery(mediaBreakpoints.xlarge)]: xl,
    [mediaQuery(mediaBreakpoints.xxlarge)]: xxl,
    [mediaQuery(mediaBreakpoints.xxxlarge)]: xxxl,
  }

  return {
    boxProps: {
      css: interStyle,
    },
    otherProps,
  }
}
