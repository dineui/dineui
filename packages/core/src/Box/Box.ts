import { jsx } from '@emotion/react'
import { FC, ReactNode } from 'react'
import { useBox, UseBoxParams } from './useBox'

export interface BoxProps extends UseBoxParams {
  children?: ReactNode | string
  as?: string
}

export const Box: FC<BoxProps> = (props) => {
  const { as = 'div', ...useBoxProps } = props
  const { boxProps, otherProps } = useBox(useBoxProps)

  return jsx(as, {
    ...boxProps,
    ...otherProps,
  })
}
