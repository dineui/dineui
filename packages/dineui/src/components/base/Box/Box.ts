import { ReactNode, forwardRef } from 'react'

import { jsx } from '@emotion/react'

import { UseBoxParams, useBox } from './useBox'

export interface BoxProps extends UseBoxParams {
  children?: ReactNode | string
  as?: string
}

export const Box = forwardRef<HTMLElement, BoxProps>((props, ref) => {
  const { as = 'div', ...useBoxProps } = props
  const { boxProps, otherProps } = useBox(useBoxProps)

  return jsx(as, {
    ref,
    ...boxProps,
    ...otherProps,
  })
})
