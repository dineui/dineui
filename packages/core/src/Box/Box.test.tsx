import { Box } from './Box'
import renderer from 'react-test-renderer'

function toJson(component: renderer.ReactTestRenderer) {
  const result = component.toJSON()
  expect(result).toBeDefined()
  expect(result).not.toBeInstanceOf(Array)
  return result as renderer.ReactTestRendererJSON
}

describe('Box component', () => {
  test('general usage', () => {
    const component = renderer.create(<Box>box</Box>)

    let tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })

  // test('style', () => {
  //   render(
  //     <Box
  //       xs={{
  //         width: 100,
  //       }}>
  //       box
  //     </Box>,
  //   )
  // })
})
