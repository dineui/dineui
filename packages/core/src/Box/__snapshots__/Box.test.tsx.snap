// Vitest Snapshot v1

exports[`Box component > general usage 1`] = `
<div
  className="emotion-0"
>
  box
</div>
`;

exports[`Box component > responsive props 1`] = `
.emotion-0 {
  width: 1px;
}

@media (min-width: 414px) {
  .emotion-0 {
    width: 2px;
  }
}

@media (min-width: 960px) {
  .emotion-0 {
    width: 3px;
  }
}

@media (min-width: 1200px) {
  .emotion-0 {
    width: 4px;
  }
}

@media (min-width: 1440px) {
  .emotion-0 {
    width: 5px;
  }
}

@media (min-width: 1920px) {
  .emotion-0 {
    width: 6px;
  }
}

@media (min-width: 2560px) {
  .emotion-0 {
    width: 7px;
  }
}

<div
  className="emotion-0"
>
  box
</div>
`;

exports[`Box component > xs props 1`] = `
.emotion-0 {
  width: 100px;
}

<div
  className="emotion-0"
>
  box
</div>
`;
